pkg/kubectl/cmd/create_role.go

```diff
@@ -255,10 +316,23 @@ func generateResourcePolicyRules(mapper meta.RESTMapper, verbs []string, resourc
 	// 2. Prevents pointing to non-existent resources.
 	// 3. Transfers resource short name to long name. E.g. rs.extensions is transferred to replicasets.extensions
 	for _, r := range resources {
-		resource, err := mapper.ResourceFor(schema.GroupVersionResource{Resource: r.Resource, Group: r.Group})
-		if err != nil {
-			return []rbac.PolicyRule{}, err
+		resource := schema.GroupVersionResource{Resource: r.Resource, Group: r.Group}
+		groupVersionResource, err := mapper.ResourceFor(schema.GroupVersionResource{Resource: r.Resource, Group: r.Group})
+		if err == nil {
+			resource = groupVersionResource
+		} else {
```

Comment:
This `else` block isn't obvious to me.  Is this trying to add a fake mapping for `userextras` so you don't have to type the group?  If that's all it is, I'd say the use-case is rare enough that I'm ok making people type it all out than special casing it like this in this command.